From 98b7fe2ad4055a7978edc28caaadd5a0a9bd97a9 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Tue, 2 Sep 2014 14:17:59 +0200
Subject: [PATCH] terminal: discard async read() errors for evdev

If read() fails on evdev devices, we deal with this in idev_evdev_hup().
It is very likely this is an async revoke, therefore, we must not abort.
Fix our io helper to discard such errors after passing them to
idev_evdev_hup(), so we don't bail out of the event loop.
---
 src/libsystemd-terminal/idev-evdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libsystemd-terminal/idev-evdev.c b/src/libsystemd-terminal/idev-evdev.c
index c93ede8dc9..9e2dc811ef 100644
--- a/src/libsystemd-terminal/idev-evdev.c
+++ b/src/libsystemd-terminal/idev-evdev.c
@@ -217,7 +217,7 @@ static int idev_evdev_io(idev_evdev *evdev) {
 
 error:
         idev_evdev_hup(evdev);
-        return r;
+        return 0; /* idev_evdev_hup() handles the error so discard it */
 }
 
 static int idev_evdev_event_fn(sd_event_source *s, int fd, uint32_t revents, void *userdata) {
